web-common/src/features/sources/modal/AddDataForm.svelte

```diff
@@ -170,127 +239,229 @@
+                    hint={property.hint}
+                    href={property.docsUrl}
+                  />
+                {/if}
+              </div>
+            {/each}
+          {/if}
+        </form>
       {:else}
-        Add data
+        <!-- Form 2: DSN -->
+        <form
+          id={dsnFormId}
+          use:dsnEnhance
+          on:submit|preventDefault={dsnSubmit}
+        >
+          {#if dsnError}
+            <SubmissionError message={dsnError} />
+          {/if}
+
+          {#each dsnProperties as property (property.key)}
+            {@const propertyKey = property.key ?? ""}
+            <div class="py-1.5">
+              <Input
+                id={propertyKey}
+                label={property.displayName}
+                placeholder={property.placeholder}
+                secret={property.secret}
+                hint={property.hint}
+                errors={$dsnErrors[propertyKey]}
+                bind:value={$dsnForm[propertyKey]}
+                alwaysShowError
+              />
+            </div>
+          {/each}
+        </form>
       {/if}
-    </Button>
+    </div>
+    <div
+      class="flex items-center justify-between space-x-2 px-6 py-4 border-t border-gray-200"
+    >
+      <Button onClick={onBack} type="secondary">Back</Button>
+      <Button disabled={submitting} form={formId} submitForm type="primary">
+        {#if isConnectorForm}
+          {#if submitting}
+            Testing connection...
+          {:else}
+            Test and Connect
+          {/if}
+        {:else}
+          Add data
+        {/if}
+      </Button>
+    </div>
+  </div>
+
+  <div class="add-data-side-panel">
+    <div>
+      <div class="text-sm leading-none font-medium mb-4">
+        Connection preview
+      </div>
+      <pre>{yamlPreview}</pre>
+    </div>
+    <div>
+      <div class="text-sm leading-none font-medium mb-4">Help</div>
+      <div
+        class="text-sm leading-normal font-medium text-muted-foreground mb-2"
+      >
+        Need help connecting to {connector.displayName}? Check out our
+        documentation for detailed instructions.
+      </div>
+      <span class="flex flex-row items-center gap-2 group">
+        <a
+          href={connector.docsUrl || "https://docs.rilldata.com/build/connect/"}
+          rel="noreferrer noopener"
+          target="_blank"
+          class="text-sm leading-normal text-primary-500 hover:text-primary-600 font-medium group-hover:underline"
+        >
+          How to connect to {connector.displayName}
+        </a>
+        <ExternalLinkIcon size="16px" color="#6366F1" />
+      </span>
+    </div>
   </div>
 </div>
+
+<style lang="postcss">
+  .add-data-layout {
+    @apply flex flex-row h-full w-full;
+  }
+  .add-data-form-panel {
+    @apply flex-1 flex flex-col min-w-0;
+  }
+  .add-data-side-panel {
+    @apply w-96 min-w-[320px] max-w-[400px] border-l border-gray-200 pl-6 flex flex-col gap-6 p-6;
+    /* FIXME: bg-sidebar-background */
```

Comment:
TODO before merge?

Reply:
Nope, the FIXME is for the next user that touches up the file when we do have `bg-sidebar-background` theme token in the system.